v6.10

Fourdee opened the pull request:
**v6.10**
(03/07/18)

**Changes / Improvements / Optimizations:**

General | ASUS TB: GLES GPU and VPU support now enabled, once Xserver is installed.

General | 'firmware-iwlwifi': Is now a pre-req to WiFi enable. Adds support for Intel based WiFi chips by default: https://github.com/Fourdee/DietPi/issues/1855

General | "net-tools" commands (ifconfig, netstat, route, ...) were replaced by modern "ip" commands (ip a, ip r, ...) within DietPi scripts and the package therefore removed from DietPi core packages: https://github.com/Fourdee/DietPi/issues/1666

General | Removed unused "/DietPi/config.txt" from non-RPi devices: https://github.com/Fourdee/DietPi/pull/1863

General | CurlFTPFS: Removed from DietPi scripts and is no longer supported. Due to lack of security, and, single digit install count (survey).

General | Timesync: DietPi will now only check for a sucessful sync once per system boot, and, again hourly/daily if set. This is to prevent excess delay of systemd-timesyncd service, once the time has already been synced.

General | Sparky SBC: Designs patch added for DSD on MPD-5 dac , new Ids added Mytek Manhatten , LH labs 1V5 2V0 ,HD-AVP/AVA IDA-8: https://github.com/sparky-sbc/sparky-test/tree/master/dsd-marantz

DietPi-Backup | Rewritten. Userdata option removed, included by default backup. Added options to edit include/exclude filters in the GUI. Existing backups (v6.9 or less) are no longer supported: https://github.com/Fourdee/DietPi/issues/1851

DietPi-Config | Soundcards (RPi): Allo Katana, now available for selection. https://github.com/Fourdee/DietPi/issues/1849

DietPi-Config | IntelGPU Driver: Installation code added: https://github.com/Fourdee/DietPi/issues/1855#issue-333150262

DietPi-Config | Networking: You can now view the sent and recieved totals for both network devices. NB: 32bit devices will reset the values after 32bit int limit is reached (roughly 4.3GB~), this is a kernel/arch limitation: https://github.com/Fourdee/DietPi/issues/1666#issuecomment-401546728

DietPi-Drive_Manager | Rewrite and improvements:
 - Now supports ROM devices (eg: DVD). NB: compatibility for DVD/CD devices relies on kernel support. Not all devices will support DVD/CD devices, and/or their filesystem format: https://github.com/Fourdee/DietPi/issues/1858
 - Resize ext4 options added: https://github.com/Fourdee/DietPi/issues/1821
 - Support for detecting and formatting non-partitioned drives
 - You can now benchmark read/write for all available mounted drives: https://github.com/Fourdee/DietPi/issues/1858
 - Now supports mounting, viewing, removal of network drives on system (nfs4, cifs). This replaces the previous options in 'dietpi-config': https://github.com/Fourdee/DietPi/issues/1858
 - Moving of DietPi user data to filesystems that do not support permission (FAT), is no longer supported and disabled: https://github.com/Fourdee/DietPi/issues/1846

DietPi-Globals | G_BACKUP: Added to globals. This will be gradually rolled out in DietPi scripts, allowing the user to create a system backup, prior to significant system changes: https://github.com/Fourdee/DietPi/issues/1871#issuecomment-400443401

DietPi-Globals | G_CHECK_USERDATA: Added to DietPi scripts: https://github.com/Fourdee/DietPi/issues/1850#issuecomment-401407996

DietPi-Services | Rsync: Added to service control. This will prevent errors if rsync is running in daemon mode, when 'dietpi-backup' is run: https://github.com/Fourdee/DietPi/issues/1869#issuecomment-399890771

DietPi-Software | Ubooquity: Now available for installation: https://dietpi.com/phpbb/viewtopic.php?f=8&t=5&p=12969#p12969 (https://github.com/Fourdee/DietPi/issues/1845#issuecomment-397447909)

DietPi-Software | Roon Extension Manager: Now available for installation. Many thanks to @JanKoudijs for contributing this great addition!: https://github.com/Fourdee/DietPi/pull/1865

DietPi-Software | Mosquitto: Now uses Debian APT installation: https://github.com/Fourdee/DietPi/issues/1868#issuecomment-399982278

DietPi-Software | Gogs: Now runs under its own user. For new installations only: https://github.com/Fourdee/DietPi/issues/1877

DietPi-Software | Xserver: 'mesa-utils-extra' now also installed by default, useful for GLES testing 'es2_info es2gears' etc.

DietPi-Software | Moode: Removed from our software lineup: https://github.com/Fourdee/DietPi/issues/1223#issuecomment-401549371

DietPi-Software | Radarr, Sonarr and Jackett: Services optimized and no longer run in debugging mode. Should improve runtime performance. Many thanks to @userdeveloper98 for this contribution! https://github.com/Fourdee/DietPi/pull/1889

DietPi-Survey | Simplified available options. You can now either Opt In, or, Opt Out and automatically have any existing data cleared. Interactive installations will be prompted to Opt In or Out during 1st run. Automated installations are Opted In by default, you can change this once setup is completed. More information on DietPi-Survey and how to change the options: https://dietpi.com/phpbb/viewtopic.php?f=8&t=20 https://github.com/Fourdee/DietPi/issues/1827#issuecomment-396005575

PREP | 'os-prober' installed by default for x86_64 devices. Ensures dual boot OSs are detected by grub. Also added a 3 second timeout to grub boot prompt, allowing OS selection: https://github.com/Fourdee/DietPi/issues/1855

**Bug Fixes:**

ASUS TB | Resolved square (broken) X11 fonts.

Raspberry Pi | Removed "initial_turbo" setting from DietPi-Config and config.txt, as it prevents CPU governor from throttling down: https://github.com/Fourdee/DietPi/issues/1836

DietPi-Drive_Manager | Resolved incorrect detection of available drives: https://github.com/Fourdee/DietPi/issues/1858

DietPi-Software | GMrender: Resolved an issue where two systems on the same network would nullify the other. Hostname is now used for the server name, UUID used is applied via DietPi generated UUID during 1st run: https://dietpi.com/phpbb/viewtopic.php?f=11&t=3900&p=12985#p12985

DietPi-Software | Apache2: Fixed a syntax error that leads to Apache logging to "/error.log" instead of "/var/log/apache2/error.log"

DietPi-Software | Nukkit: Fixed the broken download link on installation. Many thanks to @symbios24 for reporting bug and providing solution: https://github.com/Fourdee/DietPi/issues/1875

DietPi-Software | Linux software: Resolved an issue with NULL entry being displayed: https://github.com/Fourdee/DietPi/pull/1830#issuecomment-401612168

DietPi-Config | Fixen an issue, where IPv6 could not be disabled on RPi. On current kernel version it is no dedicated kernel module any more and needs to be toggled via "/boot/cmdline.txt".

**AlloGUI v9:**
- Changing the root password, no longer breaks web interface: https://github.com/Fourdee/DietPi/issues/1841
- Resolved issues with terminal leakage in the web interface: https://github.com/Fourdee/DietPi/issues/1841
- System settings: Current version is always shown, even if an update is available.
- System settings: Allo Katana, now available for selection. https://github.com/Fourdee/DietPi/issues/1849

Commit:
v6.10

+ Version codes

Commit:
v6.10

+ HW_MODEL additions/changes

MichaIng committed:
v6.10

+ DietPi-PREP | Keep timesyncd enabled, as it's new default since v6.9 and do not "rm /etc/cron.daily/ntp", as NTP is not installed anymore
+ DietPi-PREP | Remove "zip" from DietPi core packages, as Survey and Bugreport pack now via 7z
+ DietPi-PREP | Keep "APT::AutoRemove::RecommendsImportant "false";" as on new images no recommends are auto installed anyway and this allows potentially better cleanup via G_AGA. e.g. dietpi-software installed package is by chance recommend of some other installed package
+ DietPi-PREP | Only handle dietpi-ramlog if it exists. This handles a very rare error that should actually not occur, however it doesn't hurt.

MichaIng committed:
v6.10

+ DietPi-PREP | Set /tmp/dietpi-prep as working directory to potentially reduce disk I/O and increase performance

MichaIng committed:
v6.10

+ DietPi-Update | Run DietPi-Survey at the end of successful DietPi-Update

MichaIng committed:
v6.10

+ DietPi-Survey | Force interactive user choice, if survey settings file does not exist.
+ DietPi-Survey | Only write settings file, if user successfully selected choice or successfully uploaded survey file.

MichaIng committed:
v6.10

+ DietPi-PREP | dietpi.com pub host key

Fourdee committed:
Merge pull request #1834 from Fourdee/survey_tuning

DietPi-Survey | Tuning

Fourdee committed:
Merge pull request #1830 from Fourdee/PREP

DietPi-PREP | Several

Commit:
v6.10

+ Should be local vars

Commit:
Merge branch 'testing' of https://github.com/Fourdee/DietPi into testing

Commit:
v6.10

+ +DietPi-Survey | Simplified available options. You can now either Opt In, or Opt Out and automatically have any existing data cleared. Interactive installations will be prompted to Opt In or Out during 1st run. Automated installations are Opted In by default, you can change this at once setup is completed. More information on DietPi-Survey and how to change the options: https://dietpi.com/phpbb/viewtopic.php?f=8&t=20 https://github.com/Fourdee/DietPi/issues/1827#issuecomment-396005575

Commit:
v6.10

+ Minor text

MichaIng committed:
v6.10

+ DietPi-Cloudshell | Replace "net-tools" commands with "ip" and "/proc/sys/net" or "/sys/class/net/" where applicable.
+ DietPi-Cloudshell | General code rework
NB: #/sys/class/net/ values are being reset by system/kernel when they reach X size. Some sort of "cap".
	- Retest? Even on long time running dietpi.com server, values match with output of currently used "netstat -i"
	- Alternative "cat /proc/sys/net"?
	- Otherwise workaround/fail_safe by sum new value to old value, if (( $NETWORK_USAGE_TOTAL_CURRENT_RECIEVED < $total_previous_recieved ))?

MichaIng committed:
v6.10

+ DietPi-Boot | Switch from "net-tools" commands (route) to "ip" commands (ip r)

MichaIng committed:
v6.10

+ DietPi-Bugreport | Replace "net-tools" commands (ifconfig) with "ip" commands (ip a + ip r + ip l to not miss something)

MichaIng committed:
v6.10

+ tiny

MichaIng committed:
v6.10

+ tiny

MichaIng committed:
v6.10

+ DietPi-Config | Replace "ifconfig" with "ip a", add CIDR to mask function to generate e.g. 255.255.255.0 out of 192.168.0.100/**24**
+ DietPi-Config | General code improvements

MichaIng committed:
v6.10

+ DietPi-PREP | Remove "net-tools" from DietPi core packages, as those commands are replaced by modern "ip" package: https://github.com/Fourdee/DietPi/issues/1666

MichaIng committed:
v6.10

+ CHANGELOG | "net-tools" commands (ifconfig, netstat, route, ...) were replaced by modern "ip" commands (ip a, ip r, ...) within DietPi scripts and the package therefore removed from DietPi core packages: https://github.com/Fourdee/DietPi/issues/1666

Fourdee committed:
Merge pull request #1839 from Fourdee/rip_net-data

Generel | Remove "net-tools" commands in favour of "ip" and "/proc/" + "/sys/" scraping

Commit:
v6.10

+ Minor, spacing

Commit:
Merge branch 'testing' of https://github.com/Fourdee/DietPi into testing

Commit:
v6.10

+ Minor fixes/whip addition: https://github.com/Fourdee/DietPi/pull/1839#issuecomment-396438832

Commit:
v6.10

+ +AlloGUI:
- Changing the root password, no longer breaks web interface: https://github.com/Fourdee/DietPi/issues/1841
- Resolved issues with terminal leakage in the web interface: https://github.com/Fourdee/DietPi/issues/1841

Commit:
v6.10

+ Additional fixes

MichaIng committed:
v6.10

+ Whoopsie, add forgotten "eth"/"wlan" string: https://github.com/Fourdee/DietPi/issues/1666#issuecomment-396468199

MichaIng committed:
META | Add full GPLv2 license text to repo root

Commit:
v6.10

+ +DietPi-Drive_Manager | Moving of DietPi user data to filesystems that do not support permission (FAT), is no longer supported and disabled: https://github.com/Fourdee/DietPi/issues/1846

Commit:
v6.10

+ Global function for https://github.com/Fourdee/DietPi/issues/1846

Commit:
Merge branch 'testing' of https://github.com/Fourdee/DietPi into testing

Commit:
v6.10

+ https://github.com/Fourdee/DietPi/issues/1846

Commit:
v6.10

+DietPi-Config | Soundcards (RPi): Allo Katana, now available for selection. https://github.com/Fourdee/DietPi/issues/1849

Commit:
v6.10

+ katana requires latest kernel: https://github.com/Fourdee/DietPi/issues/1849

Commit:
v6.10

+DietPi-Backup | Rewritten. Userdata option removed, included by default backup. Added options to edit include/exclude filters in the GUI. Existing backups (v6.9 or less) are no longer supported: https://github.com/Fourdee/DietPi/issues/1851

Commit:
v6.10

+ Backups can now exist in userdata directory https://github.com/Fourdee/DietPi/issues/1851

Commit:
v6.10

+ Minor text

Commit:
v6.10

+ remove debug

Commit:
v6.10

+ Revert userdata location allow, breaks system if userdata location moved after the backup was done, then restored.

Commit:
v6.10

+ User notice: https://github.com/Fourdee/DietPi/issues/1851

Commit:
v6.10

+ +DietPi-Software | Ubooquity: Now available for installation: https://dietpi.com/phpbb/viewtopic.php?f=8&t=5&p=12969#p12969 (https://github.com/Fourdee/DietPi/issues/1845#issuecomment-397447909)

Commit:
v6.10

+ service https://github.com/Fourdee/DietPi/issues/1845#issuecomment-397447909

+ removal of 'c' from 'ps' command, prevents forked binaries (java) from showing the details we need.

Commit:
v6.10

+ Speed up dietpi-process tool
Old= real    0m1.067s
New = real    0m0.914s
- Removes kernel processes from list
- simple pattern match

Commit:
v6.10

+ Minor: https://github.com/Fourdee/DietPi/issues/1845#issuecomment-397447909

Commit:
v6.10

+ lightdm pre-req

Commit:
v6.10

+ Minor

MichaIng committed:
v6.10

+ DietPi-Obtain_network_details | Prefer and use only active network devices to estimate IP: https://github.com/Fourdee/DietPi/issues/1842
+ DietPi-Obtain_network_details | General code rework and speed up

MichaIng committed:
v6.10

+ DietPi-Cloudshell | Fall back to "/DietPi/dietpi/.network" to obtain IP
+ DietPi-Cloudshell | Minor code improvements

Fourdee committed:
Merge pull request #1853 from Fourdee/obtain_network_details

DietPi-Obtain_network_details | Prefer and use only active network devices to estimate IP

Commit:
v6.10

+ RPi 2 HW revision codes.

Commit:
Merge branch 'testing' of https://github.com/Fourdee/DietPi into testing

MichaIng committed:
v6.10

+ DietPi-Update | Access dietpi.com via HTTPS
+ DietPi-Update | FP_ variable naming convention
+ DietPi-Update | File path variables with static assigned well known non white space value, are safe to call without double quotes to speed up script a tiny bid.

Commit:
v6.10

+ DietPi-Software | GMrender: Resolved an issue where two systems on the same network would nullify the other. Hostname is now used for the server name, UUID used is applied via DietPi generated UUID during 1st run: https://dietpi.com/phpbb/viewtopic.php?f=11&t=3900&p=12985#p12985

Commit:
Merge branch 'testing' of https://github.com/Fourdee/DietPi into testing

Commit:
v6.10

+ RockPro64 packages

MichaIng committed:
META | Add image request template

MichaIng committed:
META | Add software request template

MichaIng committed:
META | Pull request template additions

MichaIng committed:
META | Add separate contribution guidelines file

Commit:
v6.10

+ +General | 'firmware-iwlwifi': Is now a pre-req to WiFi enable. Adds support for Intel based WiFi chips by default: https://github.com/Fourdee/DietPi/issues/1855

+ +PREP | 'os-prober' installed by default for x86_64 devices. Ensures dual boot OSs are detected by grub. Also added a 3 second timeout to grub boot prompt, allowing OS selection: https://github.com/Fourdee/DietPi/issues/1855

Commit:
Merge branch 'testing' of https://github.com/Fourdee/DietPi into testing

Commit:
v6.10

+ minor

Commit:
v6.10

+DietPi-Config | IntelGPU Driver: Installation code added: https://github.com/Fourdee/DietPi/issues/1855#issue-333150262

Commit:
v6.10

+ Covers non-dietpi-software install

Commit:
v6.10

+ optimize

Commit:
v6.10

+ minor

Commit:
v6.10

+ Whoopsie, reorder

Commit:
v6.10

+ Whoopsie ;)

Commit:
v6.10

+ Fix wlan detection: https://github.com/Fourdee/DietPi/issues/1855

Commit:
v6.10

+ Remove chromium warnings under root.

Commit:
v6.10

+ Prevent NTPD sync hanging boot during 1st run, before network established: https://github.com/Fourdee/DietPi/issues/1829#issuecomment-398273470

+ NanoPC serial root login allow: https://github.com/Fourdee/DietPi/issues/1829#issuecomment-398273470

Commit:
v6.10

+ Only run ntpd when we need it: https://github.com/Fourdee/DietPi/issues/1829

Commit:
v6.10

+ Serial required for debugging NanoPC T4: https://github.com/Fourdee/DietPi/issues/1829

Commit:
v6.10

+ Ensure we have network before running survey

Commit:
v6.10

+ Update GLOBAL_PW when changed

Commit:
v6.10

+ https://github.com/Fourdee/DietPi/issues/1829#issuecomment-398302497

MichaIng committed:
v6.10

+ DietPi-Software | Apache2: Fixed a syntax error that leads to Apache logging to "/error.log" instead of "/var/log/apache2/error.log"

MichaIng committed:
v6.10

+ DietPi-Software | Apache2: Fixed a syntax error that leads to Apache logging to "/error.log" instead of "/var/log/apache2/error.log"

MichaIng committed:
v6.10

+ DietPi-Patchfile | Fix Apache2 logging to "/error.log" instead of to intended "/var/log/apache2/error.log"

MichaIng committed:
v6.10

+ DietPi-Globals | G_CONFIG_INJECT: Minor (visual) adjustments, show correct message, when adding tab via "\t" to "$setting"

MichaIng committed:
v6.10

+ RPi | Disable "initial_turbo" setting within config.txt, as it leads to CPU governor not being able to throttle down CPU: https://github.com/Fourdee/DietPi/issues/1836

MichaIng committed:
v6.10

+ DietPi-Patchfile | Comment "initial_turbo" option from config.txt on DietPi patch

MichaIng committed:
v6.10

+ CHANGELOG | RPi: Disabled "initial_turbo" setting in config.txt, as it prevents CPU governor from throttling down

Fourdee committed:
Merge pull request #1859 from Fourdee/RPi_initial_turbo_fix

RPi | Disable "initial_turbo" setting within config.txt

Commit:
v6.10

+ Fix detection of #initial_turbo: https://github.com/Fourdee/DietPi/issues/1836#issuecomment-398557535

Commit:
Merge branch 'testing' of https://github.com/Fourdee/DietPi into testing

Commit:
v6.10

+ Fix incorrect cat listing: https://github.com/Fourdee/DietPi/issues/1858#issuecomment-398690458

+ Minor code improve

MichaIng committed:
v6.10

+ DietPi-Patchfile | RPi: Change initial_turbo comment string to prevent doubled entries on reapplied patch

MichaIng committed:
v6.10

+ DietPi-Patchfile | RPi: Just fix initial_turbo, if active (uncommented) to prevent potential double entry issues.

MichaIng committed:
v6.10

+ DietPi-Config | RPi: Disabled initial_turbo setting due to: due to: https://github.com/Fourdee/DietPi/issues/1836

MichaIng committed:
v6.10

+ DietPi-Set_Software | Move all "config.txt" adjustments into RPi check statements
+ DietPi-Set_Software | RPi: Switch from "gpu_mem_256/512/1024" to "gpu_mem"
+ DietPi-Set_Software | Use G_CONFIG_INJECT where possible for failure/duplicate safer file edit
+ DietPi-Set_Software | General code improvements

MichaIng committed:
v6.10

+ DietPi-PREP | Move "config.txt" in place only for RPi systems

MichaIng committed:
v6.10

+ RPi | "config.txt": Switch from "gpu_mem_256/512/1024" to "gpu_mem"

MichaIng committed:
v6.10

+ DietPi-Patchfile | Remove config.txt for non-RPi devices and switch from "gpu_mem_256/512/1024" to "gpu_mem" setting

MichaIng committed:
v6.10

+ CHANGELOG | Removed unused "/DietPi/config.txt" form non-RPi devices and switched from "gpu_mem_256/512/1024" settings to single "gpu_mem" setting

Commit:
v6.10

+ +DietPi-Drive_Manager | You can now benchmark read/write for all available mounted drives: https://github.com/Fourdee/DietPi/issues/1858
DietPi-Drive_Manager | Now supports mounting, viewing, removal of network drives on system: https://github.com/Fourdee/DietPi/issues/1858

Commit:
Merge branch 'testing' of https://github.com/Fourdee/DietPi into testing

MichaIng committed:
v6.10

+ CHANGELOG | Minor wording

Commit:
v6.10

+General | CurlFTPFS: Removed from DietPi scripts and is no longer supported. Due to lack of security, and, single digit install count (survey).

+ https://github.com/Fourdee/DietPi/issues/1858#issuecomment-399621291

JanKoudijs committed:
DietPi-Software | Add Roon Extension Manager

JanKoudijs committed:
Switched to ID 86

Commit:
v6.10

+ udev removal: https://github.com/Fourdee/DietPi/issues/1858

Fourdee committed:
Merge branch 'testing' into config_txt_rpi_only

Fourdee committed:
syntax fix

Fourdee committed:
Merge pull request #1863 from Fourdee/config_txt_rpi_only

RPi | Remove "config.txt" from non-RPi devices

Fourdee committed:
Merge pull request #1865 from JanKoudijs/testing

DietPi-Software | Add Roon Extension Manager

Commit:
v6.10

+ +General | Timesync: DietPi will now only check for a sucessful sync once per system boot, and, again hourly/daily if set. This is to prevent excess delay of systemd-timesyncd service, once the time has already been synced.

Commit:
Merge branch 'testing' of https://github.com/Fourdee/DietPi into testing

Commit:
v6.10

+ +DietPi-Software | Roon Extension Manager: Now available for installation. Many thanks to @JanKoudijs for contributing this great addition!: https://github.com/Fourdee/DietPi/pull/1865

+ process_tool/service control for roon ext man

Commit:
v6.10

+ Cap FS benchmark to 1GB

MichaIng committed:
v6.10

+ DietPi-Globals | G_DIETPI-NOTIFY: Speed up notifications by defining variables and functions just where they are used and execute "echo" just a single time each notification.
+ DietPi-Globals | Minor: Try to scrape "/DietPi/dietpi/.hw_model" first, fill with 0/NULL afterwards, if empty. Otherwise 0/NULL could be overwritten with empty string.

MichaIng committed:
v6.10

+ DietPi-Process_Tool | Simplify/Fix status messages by using G_DIETPI-NOTIFY as expected

MichaIng committed:
v6.10

+ G_DIETPI-NOFITY -2: Make terminal PS1 promt kill processing animation as last resort

Commit:
v6.10

+ Minor fix

Commit:
v6.10

+ +DietPi-Services | Rsync: Added to service control. This will prevent errors if rsync is running in daemon mode, when 'dietpi-backup' is run: https://github.com/Fourdee/DietPi/issues/1869#issuecomment-399890771

Commit:
v6.10

+ Syntax fix:
root@DietPi:~# sed -i '^/[[:blank:]#]*gpu_mem/d' /DietPi/config.txt
sed: -e expression #1, char 1: unknown command: `^'

Commit:
v6.10

+ Removal of old NTP system
+ Patch cleanup of installed states

Commit:
v6.10

+DietPi-Software | Mosquitto: Now uses Debian APT installation: https://github.com/Fourdee/DietPi/issues/1868#issuecomment-399982278

Commit:
v6.10

killall -w rsync &> /dev/null # :https://github.com/Fourdee/DietPi/issues/1869#issuecomment-400036370

Commit:
v6.10

+ dietpi-sync: https://github.com/Fourdee/DietPi/issues/1869#issuecomment-400036370

Fourdee committed:
Merge pull request #1867 from Fourdee/dietpi-notify-speed

DietPi-Globals | Speed up G_DIETPI-NOTIFY, let PS1 promt kill processing animation as last resort

Commit:
v6.10

+DietPi-Drive_Manager | Resize ext4 options added: https://github.com/Fourdee/DietPi/issues/1821

Commit:
v6.10

+ no partion table detect + format options:  https://github.com/Fourdee/DietPi/issues/1858

This works for me on NanoPC T4, however, once the drive is formated, system must be rebooted as /dev/sda does not exist, partprobe is useless.

Commit:
v6.10

+ Whoopsie, forgot nofail

Commit:
v6.10

+ Handle https://github.com/Fourdee/DietPi/issues/1858#issuecomment-400312970

MichaIng committed:
v6.10

+ DietPi-Software | Changed `G_AGP`/`apt-get mark` wildcard usage to `$(dpkg --get-selections package* | awk '{print $1}')`, to prevent noisy output about non-installed packages
+ DietPi-Software | Removed `Banner_Setup`, as there is always a whiptail menu or `G_DIETPI-NOTIFY 3` showing script info before
+ DietPi-Software | Always call `dietpi-survey` with $1=1, to prevent doubled whiptail menu, as this will show up automatically if no survey settings file was found (first `G_USER_INPUT=1` run)
+ DietPi-Software | Minor code improvements

MichaIng committed:
v6.10

+ DietPi-Software | Nukkit: Fix broken download link: https://github.com/Fourdee/DietPi/issues/1875

MichaIng committed:
v6.10

+ CHANGELOG | Nukkit: Fixed broken download link: https://github.com/Fourdee/DietPi/issues/1875

Commit:
v6.10

+DietPi-Software | Gogs: Now runs under its own user. For new installations only: https://github.com/Fourdee/DietPi/issues/1877

Commit:
Merge branch 'testing' of https://github.com/Fourdee/DietPi into testing

Commit:
v6.10

+ Requires home dir: https://github.com/Fourdee/DietPi/issues/1877#issuecomment-400680146

Commit:
v6.10

+DietPi-Globals | G_BACKUP: Added to globals. This will be gradually rolled out in DietPi scripts, allowing the user to create a system backup, prior to significant system changes: https://github.com/Fourdee/DietPi/issues/1871#issuecomment-400443401

Commit:
v6.10

+ minor

Commit:
v6.10

https://github.com/Fourdee/DietPi/issues/1871#issuecomment-400701756

Commit:
v6.10

+ Fix /dev//dev/X

Commit:
v6.10

+ RootFS transfer fixes: https://github.com/Fourdee/DietPi/issues/1861

Commit:
v6.10

+ Code improve

Commit:
v6.10

+ Resolve patch issue with G_USER_INPUTS=0:

[  OK  ] DietPi-Software | No changes applied for: Xserver
[ SUB2 ] DietPi-Set_software > passwords ()
[ INFO ] DietPi-Set_software | Unknown input value (). Nothing has been applied.

Commit:
v6.10

+ Whoopsie.

Commit:
v6.10

+ fix rootFS remount for init

Commit:
v6.10

+ Improve rsync service control: https://github.com/Fourdee/DietPi/issues/1869#issuecomment-401017251

Commit:
v6.10

+ One of those days...

Commit:
v6.10

 +ASUS TB | Resolved square (broken) X11 fonts.

Commit:
v6.10

Sytax

MichaIng committed:
v6.10

+ DietPi-Cloudshell | Tiny code improvement

Commit:
v6.10

+General | Sparky SBC: Designs patch added for DSD on MPD-5 dac , new Ids added Mytek Manhatten , LH labs 1V5 2V0 ,HD-AVP/AVA IDA-8: https://github.com/sparky-sbc/sparky-test/tree/master/dsd-marantz

+ ASUS TB GPU testing

Commit:
Merge branch 'testing' of https://github.com/Fourdee/DietPi into testing

Commit:
v6.10

+DietPi-Software | Xserver: 'mesa-utils-extra' now also installed by default, useful for GLES testing 'es2_info es2gears' etc.

Commit:
v6.10

+DietPi-Globals | G_CHECK_USERDATA: Added to DietPi scripts: https://github.com/Fourdee/DietPi/issues/1850#issuecomment-401407996

MichaIng committed:
v6.10

+ DietPi-Config | Fixed disabling IPv6 on RPi, since it is no kernel module any more

MichaIng committed:
v6.10

+ DietPi-Config | Fixen an issue, where IPv6 could not be disabled on RPi. On current kernel version it is no dedicated kernel module any more and needs to be toggled via "/boot/cmdline.txt".

Commit:
v6.10

+ /var/log and /tmp fixes : https://github.com/Fourdee/DietPi/issues/1858

Commit:
v6.10

+DietPi-Config | Networking: You can now view the sent and recieved totals for both network devices: https://github.com/Fourdee/DietPi/issues/1666#issuecomment-401546728

Commit:
v6.10

+DietPi-Software | Moode: Removed from our software lineup: https://github.com/Fourdee/DietPi/issues/1223#issuecomment-401549371

Commit:
v6.10

+ Patch removal, just in-case opted out

Commit:
v6.10

+ G_WHIP_INPUTBOX, treat no input/entry as a failure (allows $? != 0 to break). And minor code improve/optimize

Commit:
v6.10

+ Minor fix for current image which does not contain these.

Commit:
v6.10

+ syntax

Commit:
v6.10

+ RC

Fourdee commented:
@MichaIng 

Release candidate for release tomorrow.

Pending from my side:
- Resolve or disable rootFS transfer

Commit:
v6.10

+General | ASUS TB: GLES GPU and VPU support now enabled, once Xserver is installed.

Commit:
v6.10

+ Minor code improve

MichaIng commented:
@Fourdee 
- [x] Tiny on DietPi-PREP: Forgot to press `Start  a preview` button 😅, but just a minor anyway. Fixed: https://github.com/Fourdee/DietPi/pull/1881/commits/7e8b25c8072b502ffcff56d92489a2fb6e3259aa
- [x] Minor on G_DIETPI-NOTIFY colorizing (fixed: https://github.com/Fourdee/DietPi/pull/1881/commits/a25b67ad5b4145a263d654225a800c82ca0db170):
![colorissue](https://user-images.githubusercontent.com/28480705/42133345-c7ce3b62-7d28-11e8-9c6f-2b947c144c35.png)
  - Found the reason: Color codes > 47m are not understood by all terminals, e.g. `\e[90m` is actually bright black (gray), but to allow this within all terminal, we need to use `\e[1;30m`. Same for the animation dot color, where one appeared as blue, because 93 (bright yellow) is not understood, instead `\e[1;33m` perfectly works. Combination of bright and the 9X colors anyway do not change something. So:
![colorissue](https://user-images.githubusercontent.com/28480705/42133685-35b0ea8a-7d2e-11e8-8117-1989002e8f1e.png)
- [x] Same for DietPi-Banner colourizing: https://github.com/Fourdee/DietPi/pull/1881/commits/7a56d17d86c61b969e38dcba774144efa5f1c971
  - Could need some clean up still, don't know about this `\e[38;5;154m`, looks like RGB code? It seems to be supported by all terminals, but on Jessie VM terminal it is finally just shown as bright yellow, maybe due to limited amount of supported colours.
- [ ] DietPi-Drive_Manager still finds and allows handling of empty ROM drives. Format etc of course fails:
![rom](https://user-images.githubusercontent.com/28480705/42133736-b48ca1a0-7d2e-11e8-92af-28999c25509d.png)
- [ ] Error on formatting totally empty drive (no partitions, no UUID), due to `aDRIVE_MOUNT_TARGET[$INDEX_INIT_CURRENT_DEVICE]=${aDRIVE_MOUNT_SOURCE[$INDEX_INIT_CURRENT_DEVICE]}` and `rm -R $FORMAT_PREVIOUS_MOUNT_TARGET`.
  - You already faced this here: https://github.com/Fourdee/DietPi/issues/1858#issuecomment-400312970
- [ ] DietPi-Survey upgrade, **PR is up, WIP: https://github.com/Fourdee/DietPi/pull/1884**
  - What you thinks about https://github.com/Fourdee/DietPi/issues/1827#issuecomment-397304683 ? For the first it is less flexible, than doing your approach with own array index for every upload file, because the connections between e.g. used device and installed software is lost (could be interesting as well if certain software/setting is mostly used on certain device, or if certain software is mostly installed together with certain other software), but it is way faster, if we are for now just interested in the overall counts. Using `grep` on the files still allows to check for combination of values, but is then due to speed again not useful for automatic report page creation.
  - Secondly: https://github.com/Fourdee/DietPi/issues/1827#issuecomment-399689518. You want to keep survey upload count, or remove it? I will add the automation count.

MichaIng committed:
v6.10

+ DietPi-Globals | G_DIETPI-NOTIFY: Fix color codes, to be supported by all kind of terminals

MichaIng committed:
v6.10

+ DietPi-Banner | Fix color codes, to be supported by all kind of terminals

MichaIng committed:
v6.10

+ DietPi-PREP | Go back to tmp working dir after loading dietpi-globals, as this contains "cd $HOME"

Fourdee commented:
@MichaIng 

> ROM

I'll get this implemented properly in v6.11, needs more time than I have available today.

note to self:
```
		#Set flags for drives
		for ((i=0; i<${#aDRIVE_MOUNT_SOURCE[@]}; i++))
		do

			#ROM (eg: cd)
			if [[ $aDRIVE_MOUNT_SOURCE == '/dev/sr'* ]]; then

				aDRIVE_ISROM[$i]=1

			fi

		done

#or in menu
if [[ $aDRIVE_MOUNT_SOURCE == '/dev/sr'* ]]; then
arrr this be a CD rom!
```

> Error on formatting totally empty drive (no partitions, no UUID), due to 

Did you get the reboot prompt? Once rebooted, the drive should be detected correctly? Tried partprobe and various items, but it seems only a reboot reloads basic level drive info.

Commit:
v6.10

+ Minor code optimize
+ Minor formatting

Commit:
Merge branch 'testing' of https://github.com/Fourdee/DietPi into testing

MichaIng committed:
v6.10

+ DietPi-Survey | Further simplification: Just two modes ($1=1 non-interactive, else interactive), opt choices = 0/1 instead of 0/2
+ DietPi-Survey | Silently fail in non-interactive mode
+ DietPi-Survey | Only show successful purge message in interactive mode

MichaIng commented:
@Fourdee 
>Did you get the reboot prompt? Once rebooted, the drive should be detected correctly? Tried partprobe and various items, but it seems only a reboot reloads basic level drive info.

Somehow I don't get the reboot prompt. But as said, this is easily fixable. I didn't touch drive_manager due to your ongoing work on this, but can try to fix it.

What do you think about in case move release for tomorrow? I also want to have this survey file adjustment finished, but wanted to wait for your thoughts about this 😉. I can do that until the night, but until tomorrow we can surely fix the left drive_manager quirks. "Scan-on-reboot" excluded => v6.11

Commit:
v6.10

+ Auto length scaling of category items '-'

+DietPi-Software | Linux software: Resolved an issue with NULL entry being displayed: https://github.com/Fourdee/DietPi/pull/1830#issuecomment-401612168

Fourdee commented:
> What do you think about in case move release for tomorrow?

@MichaIng 

Yep agree 👍 Tomorrow it is.

Fourdee commented:
REF: https://twitter.com/DietPi_/status/1013438979792875521

Commit:
v6.10

+ Remove debugging

Commit:
v6.10

+ Whoopsie!

Fourdee commented:
>  I didn't touch drive_manager due to your ongoing work on this, but can try to fix it.

@MichaIng 

Yep, go for it, i'll resolve the conflicts if any 👍

Commit:
v6.10

+ Part uuid for RPi's https://github.com/Fourdee/DietPi/issues/1858

Commit:
v6.10

# 	Xserver (ASUSTB GPU) reinstall

Commit:
v6.10

+ Tweaks and fixes.

Fourdee commented:
New visual style:

Note to self:
- Get max char length of whip array (index 1/odd only, ignore 0/even), then set to length of -
![image](https://user-images.githubusercontent.com/7077026/42136816-0f03d8da-7d5a-11e8-8d20-6bf03e1b4ef8.png)

Commit:
v6.10

+ New visual style for whip: https://github.com/Fourdee/DietPi/pull/1881#issuecomment-401619802
+ General code improve

MichaIng committed:
v6.10

+ DietPi-Patch_file | Fix dietpi-survey command argument, settings file and DietPi-Version (increase to target version before uploading)

Commit:
v6.10

+ Minor, simple scaling until char count for both whiptail menu entries are calculated.

MichaIng committed:
v6.10

+ DietPi-Survey | Force interactive mode, if invalid opt-in/out setting is found
+ DietPi-Survey | Switch survey file to source-able format to speed up automated survey report web page generation
+ DietPi-Survey | Minor wording fix

MichaIng committed:
v6.10

+ DietPi-Survey | Syntax

MichaIng committed:
v6.10

+ General | RPi: Revert "gpu_mem" to "gpu_mem_XXX" as it is more failure safe

MichaIng committed:
v6.10

+ General | RPi: Revert "gpu_mem" to "gpu_mem_XXX" as it is more failure safe

MichaIng committed:
v6.10

+ General | RPi: Revert "gpu_mem" to "gpu_mem_XXX" as it is more failure safe

MichaIng committed:
v6.10

+ General | RPi: Revert "gpu_mem" to "gpu_mem_XXX" as it is more failure safe

MichaIng committed:
v6.10

+ General | RPi: "config.txt" needs to be removed from /boot as well

MichaIng committed:
v6.10

+ Typo

MichaIng committed:
v6.10

+ DietPi-Drive_Manager | Re-add partition type buttons text
+ DietPi-Drive_Manager | Re-add "defaults" and "x-systemd.automount" to fstab entries
+ DietPi-Drive_Manager | Add fsck flag =1 to RootFS, to allow scan on reboot
+ DietPi-Drive_Manager | Fix: Drives without partition got mount_target=mount_source and mount_target for removed after format, which lead to error on mounting the drive until reboot.
+ DietPi-Drive_Manager | Minor code and wording

Fourdee committed:
Merge pull request #1885 from Fourdee/revert_gpu_mem

General | RPi: Revert "gpu_mem" to "gpu_mem_XXX" as it is more failure safe

Commit:
v6.10

+ Auto length for '●─', based on max length of all coloum enties.

Commit:
v6.10

+DietPi-Drive_Manager | Now supports ROM devices (eg: DVD): https://github.com/Fourdee/DietPi/issues/1858

MichaIng committed:
v6.10

+ DietPi-Survey | Patch: Whoopsie, values need to be written back to settings file

Commit:
v6.10

+ Minor info

Fourdee committed:
exclude devtmpfs 

exclude devtmpfs

Commit:
v6.10

+ Create a partition for non-partitioned devices during format:  https://github.com/Fourdee/DietPi/issues/1858

Commit:
v6.10

+ Improve formatting, Zero all available partitions aswell: https://github.com/Fourdee/DietPi/pull/1887

Fourdee committed:
Merge pull request #1884 from Fourdee/dietpi-survey

DietPi-Survey | Collected data adjustments, source-able survey file and further simplification

Commit:
v6.10

+ Revert/remove partition formating during partition table format: https://github.com/Fourdee/DietPi/commit/2747fbeb522fa6b8fd3c707d4d84b1139f31c09e#commitcomment-29571995

MichaIng commented:
Investigating, when I wanted to opt in DietPi-Survey:
```
root@VM-Jessie:~# /DietPi/dietpi/dietpi-survey
[  OK  ] Root access verified.
/DietPi/dietpi/func/dietpi-globals: line 558: ((: 551 0 47 11 27 24 27 26 27 44 33 19 29 46 30 30 27 27 27 43 0 37 > ( 120 - 4 ) : syntax error in expression (error token is "0 47 11 27 24 27 26 27 44 33 19 29 46 30 30 27 27 27 43 0 37 > ( 120 - 4 ) ")
[  OK  ] DietPi-Survey | Connection test: dietpi.com
[  OK  ] DietPi-Survey | Successfully sent survey data
```

@Fourdee 
Double quotes missing to keep line breaks within string:  `done <<< "$character_count_per_line"`
**€: Fixed: https://github.com/Fourdee/DietPi/commit/d7d5e3f5c46c8ea6ecf3f3662307cedeb01e2cef**

As it has UI, make `dietpi-survey` part of dietpi-* aliases, so path doesn't need to be typed in terminal?

MichaIng committed:
v6.10

+ DietPi-Globals | Fix missing double quotes

Commit:
Revert "v6.10"

This reverts commit a4daf2f353858db3355c8b58f73a8628cf0c50c2.

MichaIng commented on dietpi/patch_file:
```diff
@@ -725,20 +709,128 @@ _EOF_
+
+			fi
+			#-------------------------------------------------------------------------------
+			#Removal of NTP from dietpi-software
```
@Fourdee 
curlftpfs (index 2) needs to be reset as well?

MichaIng committed:
v6.10

+ DietPi-Globals | Add "G_HW_ARCH_DESCRIPTION" to global variables, as used in "dietpi-obtain_hw_model" and to be used in "dietpi-survey"

MichaIng committed:
v6.10

+ DietPi-Survey | Use new global "G_HW_ARCH_DESCRIPTION"

Commit:
v6.10

+ Automatically handle/detect if drive needs partition table creation, if it does not have one currently.

Commit:
Merge branch 'testing' of https://github.com/Fourdee/DietPi into testing

Commit:
v6.10

+ cleaner code with aDRIVE_ISPARTITIONTABLE

Fourdee commented:
@MichaIng 

I'am nearly done for the day, lets move this back to tomorrow https://twitter.com/DietPi_/status/1013915130801868801

> As it has UI, make dietpi-survey part of dietpi-* aliases, so path doesn't need to be typed in terminal?

Yep, go for it 👍 

> Double quotes missing to keep line breaks within string: done <<< "$character_count_per_line"
€: Fixed: d7d5e3f

Apologies, think I did that eariler

Commit:
v6.10

+ Formatting: Code/info/sanity improve

Commit:
v6.10

+ Minor info

MichaIng committed:
v6.10

+ DietPi-Globals | Add "dietpi-survey" to global aliases
+ DietPi-Globals | Add "DietPi Core Team" to github issue creator, in case
+ DietPi-Globals | Add bug report ID only, if it was actually sent, add kernel version and adjust according to current GitHub template

MichaIng commented:
@Fourdee 
Done + minor GitHub ticket creator tuning: https://github.com/Fourdee/DietPi/commit/6c43c43d04ff406403fb5474aef564275d3407b2

Jep, I think it's best to move it to tomorrow. This drive manager turns out a tricky one 😉. Have a nice sleep 👍.

userdeveloper98 commented:
@Fourdee  @MichaIng 
Hi there guys, I would like to contribute a little bit to your project, could you please grant me write access to the repository ?

--
I've created a branch for Pull Request, but I can't push it :(

MichaIng commented:
@userdeveloper98 
Go "pull requests" > click "compare across forks" and then you should be able to create pull requests from your own repository to be pushed into Fourdee/testing branch.

Contributions are always welcome of course, but please understand, that @Fourdee will not directly add you as official contributor, I guess, although maybe there is a way to do that with access limited, to edit new branches, just to skip handling an own repo?

MichaIng committed:
Merge branch 'testing' into drive_manager

MichaIng committed:
v6.10

+ Tiny alignment

MichaIng committed:
v6.10

+ use "uname -v" instead of "uname -r" to get full kernel version.

MichaIng committed:
v6.10

+ spelling

userdeveloper98 commented:
@MichaIng 
Thanks for reply, I'll try "forks workaround" ... 👍 

> I guess, although maybe there is a way to do that with access limited, to edit new branches, just to skip handling an own repo?

The branch protection may be enabled and no changes will be possible without Pull Requests, except Admin of sure :)

MichaIng committed:
v6.10

+ DietPi-Boot | "tee" output to log file instead of redirect to preserve/show logs in terminal as well

MichaIng committed:
v6.10

+ DietPi-fs_partition_resize | Minor coding enhencement

MichaIng commented:
@userdeveloper98 
It is not really a workaround but common practice. Should work fine 😀, did this myself for some months.
Short cut is to browse testing repo to the file you want to edit, click edit button will directly create a new branch on you own repo and offers to automatically create PR then to Fourdee/testing branch.

@Fourdee 
https://github.com/Fourdee/DietPi/blob/testing/rootfs/var/lib/dietpi/services/fs_partition_resize.sh#L66
This does only apply for mmc devices but not sdX? Or are the latter covered by one prior command?

userdeveloper98 committed:
Fixed ExecStart for Radarr, Sonarr and Jackett.

Running mono with debug option affects the application performance.

MichaIng committed:
v6.10

+ DietPi-CPUinfo | Minor coding enhancements

MichaIng committed:
v6.10

+ DietPi-PostBoot | Minor coding

MichaIng committed:
v6.10

+ DietPi-Create_MySQL_DB | Minor coding

MichaIng committed:
v6.10

+ DietPi-LED_Control | Minor coding

Fourdee commented:
@MichaIng 

> https://github.com/Fourdee/DietPi/blob/testing/rootfs/var/lib/dietpi/services/fs_partition_resize.sh#L66
This does only apply for mmc devices but not sdX? Or are the latter covered by one prior command?

Good spot 👍 Needs fixing. No ```p``` for ```sdX``` based drives.

MichaIng committed:
v6.10

DietPi-Optimal_MTU | Minor coding

MichaIng commented:
@Fourdee 
Hehe, has some benefits so have a casual look into our files. I am currently doing a bid minor coding enhancements, double braces, grep -q, $(<file) instead of $(cat file) and stuff like that.

Found `DietPi-Optimal_MTU`. I don't understand the background or why ping shows the error for large MTUs, but found my optimal MTU of `1492` on google.com and dietpi.com.
Maybe we can implement this into `dietpi-config` by times?

MichaIng committed:
v6.10

+ DietPi-Run_ntpd | Tiny

Commit:
v6.10

+ cat's and tweaks: https://github.com/Fourdee/DietPi/issues/1858

Commit:
Merge branch 'testing' of https://github.com/Fourdee/DietPi into testing

MichaIng committed:
v6.10

+ DietPi-CPU_set | Minor coding

Fourdee committed:
Merge pull request #1889 from userdeveloper98/enhancement/mono_param_fix

#Fixed ExecStart for Radarr, Sonarr and Jackett.

Fourdee commented:
@MichaIng 

> Maybe we can implement this into dietpi-config by times?

Yep, for v6.11 👍

Commit:
v6.10

+DietPi-Software | Radarr, Sonarr and Jackett: Services optimized and no longer run in debugging mode. Should improve runtime performance. Many thanks to @userdeveloper98 for this contribution! https://github.com/Fourdee/DietPi/pull/1889

Commit:
Merge branch 'testing' of https://github.com/Fourdee/DietPi into testing

Fourdee committed:
Merge pull request #1887 from Fourdee/drive_manager

DietPi-Drive_Manager | Fixes

Commit:
v6.10

+ Syntax fix + minor tweaks

Commit:
v6.10

+ resolve https://github.com/Fourdee/DietPi/pull/1887#issuecomment-402203501

Fourdee committed:
Merge pull request #1888 from Fourdee/boot_messages

DietPi-Boot | "tee" output to log file instead of redirect to preserve/show logs in terminal as well

Commit:
v6.10

+ https://github.com/Fourdee/DietPi/pull/1887#issuecomment-402208160

Commit:
v6.10

+ RC v2

Fourdee commented:
@MichaIng 

Appears my opted in SBC's are now opted out during recent patch:
```
root@DietPi:~# cat /DietPi/dietpi/.dietpi-survey
0
5
```

---


Sys1:
```
root@DietPi:~# cat /DietPi/dietpi/.dietpi-survey
0
15
root@DietPi:~# nano /DietPi/dietpi/.dietpi-survey
root@DietPi:~# nano /DietPi/dietpi/.dietpi-survey^C
root@DietPi:~#  cat /DietPi/dietpi/.dietpi-survey
1
15
```

sys 2:
```
root@DietPi:~# nano /DietPi/dietpi/.dietpi-survey
root@DietPi:~# cat
^C
root@DietPi:~# cat /DietPi/dietpi/.dietpi-survey
2
5

```

```dietpi-update -1```

sys 1 issue
```
root@DietPi:~# cat /DietPi/dietpi/.dietpi-survey
0
15
```

Should not affect end users, only multiple patches of 6.10, either way, I'll fix.

MichaIng committed:
v6.10

+ DietPi-Drive_Manager | Comment debug message

MichaIng commented:
@Fourdee 
Jep, when using `dietpi-update -1` this is done. Hmm we could scrape /DietPi/dietpi/.version to get real version. I will add it.

Fourdee commented:
@MichaIng 

Maybe we should save the survey version in the save file for easy patching?
```
cat << _EOF_ > $fp_temp
$opted_in
$survey_sentcount
$survey_version
_EOF_
```

MichaIng committed:
v6.10

+ DietPi-Survey | Patch: Ensure that users don't get silently opted out, if using "dietpi-update -1"

MichaIng commented:
@Fourdee 
Also an idea, but I made commit already, which is working fine to keep correct opt status: https://github.com/Fourdee/DietPi/commit/da50ea2f4c0c5e794bdc6fdf6b9d08be594def6a

Fourdee commented:
@MichaIng 

Thanks 👍 

Ok, few hours of final general testing, then i'll merge if no issues found.

---

Edit, is the new survey code added to server and as you want it?

MichaIng commented:
@Fourdee 
Jep, doing the same here 🙂.

€:
- `[ INFO ] DietPi-Bugreport | Lines required for Whiptail box and its contents (y=61, z=0), exceeds screen`
  - Remove debug messages. Actually for the future I would like to have `G_DIETPI-NOTIFY 4 message` => `(( $G_DEBUG_MODE ))` && `[ DEBUG] message`. We could leave debug messages at critical spots in place then and also make them available to end user for debugging, G_DIETPI-NOTIFY should just exit fast (check at beginning of function) to no have much delay.
- software index 2 status needs be be reset as well?

MichaIng committed:
v6.10

+ DietPi-Bugreport | Minor format issue of report file view

Fourdee commented:
Note to self:
- Move all v6.9 BR's to a archive folder

Fourdee commented:
@MichaIng 

> Remove debug messages. Actually for the future I would like to have G_DIETPI-NOTIFY 4 message

The main reason for leaving them in, is in a hope, that if the user goes "why doesn't this fit, i'll send a bug report and git ticket", they may see the info, prior to that.

Commit:
v6.10

+ enable AmiBerry TB testing

Fourdee commented:
Ok, lets release.

No major dev changes in the next few days, reserve it for any hotfixes, if they are required. 

@MichaIng 

Great work as always 👍 🥇

MichaIng commented:
Jep, I agree. Can't wait to see how fast the update is applied according to survey report :smiley:.
